lib/carrierwave/bombshelter.rb

```diff
@@ -59,14 +59,24 @@ def check_pixel_dimensions!(image)
     end
 
     def get_file(file)
-      case file
-      when CarrierWave::Storage::Fog::File
-        file.url
-      when CarrierWave::SanitizedFile, CarrierWave::Uploader::Base
-        get_file(file.file)
-      else
-        file
+      return file.url if in_fog?(file)
+      return get_file(file.file) if is_wrapped?(file)
+      file
+    end
+
+    def in_fog?(file)
+      defined?(CarrierWave::Storage::Fog) &&
+        file.is_a?(CarrierWave::Storage::Fog::File)
+    end
+
+    def is_wrapped?(file)
+      file_wrappers.reduce(false) do |acc, wrapper|
```

Comment:
It's better to use [`any?`](http://ruby-doc.org/core-2.3.0/Array.html#method-i-any-3F) here